modules/user_data/templates/start_graphdb.sh.tpl

```diff
@@ -14,130 +14,56 @@ imds_token=$( curl -Ss -H "X-aws-ec2-metadata-token-ttl-seconds: 300" -XPUT 169.
-    # read https://github.com/oogali/ebs-automatic-nvme-mapping/blob/master/README.md, for more information
-    real_device=$(nvme id-ctrl --raw-binary $volume | cut -c3073-3104 | tr -s ' ' | sed 's/ $//g')
-    if [ "$device_mapping_full" = "$real_device" ] || [ "$device_mapping_short" = "$real_device" ]; then
-      graphdb_device="$volume"
-      break
-    fi
-  done
-
-  if [ -n "$graphdb_device" ]; then
-    break
-  fi
-  sleep 5
-done
-
-# create a file system if there isn't any
-if [ "$graphdb_device: data" = "$(file -s $graphdb_device)" ]; then
-  mkfs -t ext4 $graphdb_device
-fi
 
-disk_mount_point="/var/opt/graphdb"
+# Defining variables by interpolating Terraform variables
 
-# Check if the disk is already mounted
-if ! mount | grep -q "$graphdb_device"; then
-  echo "The disk at $graphdb_device is not mounted."
+region="${region}"
+name="${name}"
+device_name="${device_name}"
 
-  # Create the mount point if it doesn't exist
-  if [ ! -d "$disk_mount_point" ]; then
-    mkdir -p "$disk_mount_point"
-  fi
+backup_schedule="${backup_schedule}"
+backup_retention_count="${backup_retention_count}"
+backup_bucket_name="${backup_bucket_name}"
 
-  # Add an entry to the fstab file to automatically mount the disk
-  if ! grep -q "$graphdb_device" /etc/fstab; then
-    echo "$graphdb_device $disk_mount_point ext4 defaults 0 2" >> /etc/fstab
-  fi
+ebs_volume_type="${ebs_volume_type}"
+ebs_volume_size="${ebs_volume_size}"
+ebs_volume_iops="${ebs_volume_iops}"
+ebs_volume_throughput="${ebs_volume_throughput}"
+ebs_kms_key_arn="${ebs_kms_key_arn}"
 
-  # Mount the disk
-  mount "$disk_mount_point"
-  echo "The disk at $graphdb_device is now mounted at $disk_mount_point."
-else
-  echo "The disk at $graphdb_device is already mounted."
-fi
+zone_dns_name="${zone_dns_name}"
+zone_id="${zone_id}"
 
-# Ensure data folders exist
-mkdir -p $disk_mount_point/node $disk_mount_point/cluster-proxy
+jvm_max_memory="${jvm_max_memory}"
+resource_name_prefix="${resource_name_prefix}"
 
-# this is needed because after the disc attachment folder owner is reverted
-chown -R graphdb:graphdb $disk_mount_point
+GRAPHDB_CONNECTOR_PORT=""
 
-# Register the instance in Route 53, using the volume id for the sub-domain
+# Search for an available EBS volume to attach to the instance. If no volume is found - create new one, attach, format and mount the volume.
+./opt/helper-scripts/ebs_volume.sh \
+ --name ${name} \
+ --ebs_volume_type ${ebs_volume_type} \
+ --ebs_volume_throughput ${ebs_volume_throughput} \
+ --ebs_kms_key_arn ${ebs_kms_key_arn} \
+ --ebs_volume_size ${ebs_volume_size} \
+ --ebs_volume_iops ${ebs_volume_iops} \
+  --device_name ${device_name}
 
-subdomain="$( echo -n "$volume_id" | sed 's/^vol-//' )"
-node_dns="$subdomain.${zone_dns_name}"
+# Register the instance in Route 53, using the volume id for the sub-domain. Capturing the node_dns variable to be used in the rest of the script.
+node_dns=$(./opt/helper-scripts/register_route53.sh \
```

Comment:
At the end of this script there's this:
```shell
echo "Instance registered in Route 53 with DNS name:"
echo "$node_dns"
```
so, you'll have a lot more than just the DNS name.
Since, you have the zone name and the volume id, you can construct the FQDN. Otherwise, the script should echo just the FQDN.